runtime/context_test.go

```diff
@@ -0,0 +1,28 @@
+package runtime_test
+
+import (
+	"github.com/gengo/grpc-gateway/runtime"
+	"google.golang.org/grpc/metadata"
+
+	"golang.org/x/net/context"
+	"net/http"
+	"testing"
+)
+
+func TestAnnotateContext(t *testing.T) {
+	ctx := context.Background()
+
+	request, _ := http.NewRequest("GET", "http://localhost", nil)
+	request.Header = http.Header{}
+	annotated := runtime.AnnotateContext(ctx, request)
+	if annotated != ctx {
+		t.Errorf("Annotation did not return proper context: %v != %v", ctx, annotated)
+	}
+	request.Header.Add("X-Grpc-Metadata-FooBar", "Value1")
+	request.Header.Add("X-Grpc-Metadata-Foo-BAZ", "Value2")
+	annotated = runtime.AnnotateContext(ctx, request)
+	md, ok := metadata.FromContext(annotated)
+	if !ok || len(md) != 2 || md["Foobar"] != "Value1" || md["Foo-Baz"] != "Value2" {
```

Comment:
Split the check into three checks.
1. `!ok` and length check
2. `md["Foobar"]`
3. `md["Foo-Baz"]`